rpcs3/Emu/RSX/Program/ProgramStateCache.cpp

```diff
@@ -606,6 +606,30 @@ fragment_program_utils::fragment_program_metadata fragment_program_utils::analys
 	const auto instBuffer = ptr;
 	s32 index = 0;
 
+	// Find the start of the program
+	while (true)
+	{
+		const auto inst = v128::loadu(instBuffer, index);
+
+		const u32 opcode = (inst._u32[0] >> 16) & 0x3F;
+		if (opcode)
```

Comment:
Technically I believe you can test against all zeroes. Because this was the way GCM was handling unaligned shader addresses, by padding the gap with zeroes.
I will merge as is though.